### PR TITLE
feat: improve example files handling in examples scripts

### PR DESCRIPTION
**Pull request by anshlachamb**
- All example files are now uploaded on the server side and the paths are uploaded accordingly, the path will always be the correct one either in local or docker configuration
- Exception made for flute_psd.txt this file is needed only locally as we're using native python functions to read it and not DPF Operators

- side modification: "abort_on_example_error": True, added in conf.py to ensure that the failure of one example generation leads to the general failure of the PR

### CODE REVIEW

File: src/ansys/sound/core/examples_helpers/download.py [hunk 1]
```diff
@@ -93,17 +94,18 @@ def _retrieve_file(url, filename, _test=False):  # pragma no cover
     return local_path
 
 
-def _download_file(filename, directory=None, _test=False):  # pragma no cover
+def _get_file_path_in_dpf_server(
```
Comment (a-bouth) [inline src/ansys/sound/core/examples_helpers/download.py:97]: @anshlachamb I feel that the name of the function is a bit incomplete as it also downloads the file (and all the other methods are named `download_flute, download_something` etc

What do you think of something like `_download_file_and_retrieve_path_in_dpf_server` ? A bit verbose but maybe more accurate ?

File: src/ansys/sound/core/examples_helpers/download.py [hunk 2]
```diff
@@ -93,17 +94,18 @@ def _retrieve_file(url, filename, _test=False):  # pragma no cover
...
         local_path = _retrieve_file(url, filename, _test)
-
-        # In case of CI/CD pipelines
-        port_in_env = os.environ.get("ANSRV_DPF_SOUND_PORT")
-        if port_in_env is not None:
-            return "C:\\data\\" + filename
-        else:
-            return local_path
+        # upload file to DPF server,
+        # so that we are independent on the local configuration (docker or not)
```
[the remark this answers is not in the document]
Comment (anssmolla) [reply src/ansys/sound/core/examples_helpers/download.py:106]: please do not mention docker here, it does not make sense

File: src/ansys/sound/core/examples_helpers/download.py (hunk 1 above)
Comment (anssmolla) [reply src/ansys/sound/core/examples_helpers/download.py:97]: I agree, the name is not very explicit
We should clarify the purpose of this function in its documentation

File: src/ansys/sound/core/examples_helpers/download.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anshlachamb) [reply src/ansys/sound/core/examples_helpers/download.py:106]: done

File: src/ansys/sound/core/examples_helpers/download.py (hunk 1 above)
Comment (anshlachamb) [reply src/ansys/sound/core/examples_helpers/download.py]: done

**Review by a-bouth — DISMISSED** [review]
looks good to me, merry christmas 🎅

**Comment by anshlachamb** [comment]
Example 6 broken. I set this comment to block the PR

**Review by anssmolla — DISMISSED** [review]

**Review by a-bouth — APPROVED** [review]

**Review by anssmolla — APPROVED** [review]

**Pull request merged**